mission_control/navigator_missions/nav_missions/detect_deliver.py

```diff
@@ -79,58 +82,84 @@ def circle_search(self):
-        if self.resp.found:
-            self.found_shape = self.resp.shapes.list[0]
-        defer.returnValue(self.resp.found)
+    def get_aligned_pose(self, enupoint, enunormal):
+        aligned_position = enupoint + self.shoot_distance_meters * enunormal  # moves x meters away
+        angle = np.arctan2(-enunormal[0], enunormal[1])
+        aligned_orientation = trns.quaternion_from_euler(0, 0, angle)  # Align perpindicular
+        return (aligned_position, aligned_orientation)
 
-    def get_aligned_pos(self):
-        self.aligned_position = self.enupoint + self.shoot_distance_meters * self.enunormal  # moves x meters away
-        angle = np.arctan2(-self.enunormal[0], self.enunormal[1])
-        self.aligned_orientation = trns.quaternion_from_euler(0, 0, angle)  # Align perpindicular
+    def get_shape_pos(self, normal_res, enu_cam_tf):
+        enunormal = enu_cam_tf.transform_vector(navigator_tools.rosmsg_to_numpy(normal_res.normal))
+        enupoint = enu_cam_tf.transform_point(navigator_tools.rosmsg_to_numpy(normal_res.closest))
+        return (enupoint, enunormal)
 
     @txros.util.cancellableInlineCallbacks
-    def get_normal(self):
+    def get_normal(self, shape):
         req = CameraToLidarTransformRequest()
-        req.header = self.found_shape.header
+        req.header = shape.header
         req.point = Point()
-        req.point.x = self.found_shape.CenterX
-        req.point.y = self.found_shape.CenterY
-        rect = self._bounding_rect(self.found_shape.points)
+        req.point.x = shape.CenterX
+        req.point.y = shape.CenterY
+        rect = self._bounding_rect(shape.points)
         req.tolerance = int(min(rect[0]-rect[3],rect[1]-rect[4])/2.0)
-        self.normal_res = yield self.cameraLidarTransformer(req)
-        if not self.normal_res.success:
-            defer.returnValue(False)
-        if not self.normal_is_sane(self.normal_res.normal):
-            self.normal_res.success = False
-            self.normal_res.error = "UNREASONABLE NORMAL"
-            defer.returnValue(False)
-        enu_cam_tf = yield self.navigator.tf_listener.get_transform('/enu', '/'+req.header.frame_id)
-        self.shooter_baselink_tf = yield self.navigator.tf_listener.get_transform('/base_link','/shooter')
-        self.enunormal = enu_cam_tf.transform_vector(navigator_tools.rosmsg_to_numpy(self.normal_res.normal))
-        self.enupoint = enu_cam_tf.transform_point(navigator_tools.rosmsg_to_numpy(self.normal_res.closest))
-        defer.returnValue(True)
+        normal_res = yield self.cameraLidarTransformer(req)
+        if not self.normal_is_sane(normal_res.normal):
+            normal_res.success = False
+            normal_res.error = "UNREASONABLE NORMAL"
+        defer.returnValue(normal_res)
 
     def normal_is_sane(self, vector3):
          return abs(navigator_tools.rosmsg_to_numpy(vector3)[1]) < 0.2
```

Comment:
Small change, but it would probably be better to put any threshold values like this at the top of the file in a variable, just in case you want to change them later. 

However, if you think this value will be fairly constant, then it is fine.